src/components/panel/Panel.tsx

```diff
@@ -1,155 +1,33 @@
-      return null
-    }
-
-    if (entrypointType === 'collection') {
-      return 'collection'
-    }
-    if (entrypointType === 'manifest') {
-      return 'manifest'
-    }
-
-    return null
-  }
-
-
-  async function readData(panelConfig: PanelConfig, documentType: string) {
-    if (!panelConfig || Object.keys(panelConfig).length === 0) return
-
-    if (error) return
-    let response
-
-    // read Document data
-    response = await request(panelConfig.entrypoint.url)
-
-    if (!response.success) {
-      setError(response.message)
-      return
-    }
-
-    const documentData = documentType === 'collection' ? response.data as Collection : response.data as Manifest
-
-    // read Manifest data
-    response = await getManifestData(documentData, documentType)
-    if (!response.success) {
-      setError(response.message)
-      return
-    }
-
-    const manifestData = response.data as Manifest
-
-    // read item data
-    if (!manifestData.sequence || manifestData.sequence.length === 0) {
-      setError ('The items of manifest ' + (manifestData.label ?? 'unknown') + ' are not defined or are empty!!')
-      return
-    }
-
-    response = await getItemData(manifestData)
-    if (!response.success) {
-      setError(response.message)
-      return
-    }
-
-    const itemData = response.data as Item
-
-    if (!isItemContentValid(itemData)) {
-      setError('Content objects are not defined for this item')
-      return
-    }
-
-    await assignContentTypes(itemData.content)
-    const contentUrl = itemData.content[activeContentTypeIndex]?.url ?? null
-
-    if (!contentUrl) {
-      setError('No content URL found.')
-      return
-    }
-    response = await request<string>(contentUrl)
-
-    if (!response.success) {
-      setError(response.message)
-      return
-    }
-
-    setText(response.data)
-
-    setLoading(false)
-  }
-
-  useEffect(() => {
-    const documentType = getDocumentType(panelConfig)
-    if (!documentType) return
-
-    readData(panelConfig, documentType)
-  }, [panelConfig, activeContentTypeIndex])
-
+const Panel: FC<PanelProps> = ({ panelId }) => {
+  // TODO: one should get a potential error regarding this panel from probably panels wrapper or a global state of error
+  //        and display the respective error component
+  const [error] = useState<boolean | string>(false)
 
   if (error) {
     return <ErrorComponent message={error} />
   }
 
-  if (loading) {
-    return <div> Loading the data of this panel...</div>
-  }
-
   return (
-    <div className="panel t-flex t-flex-col t-w-[600px] t-ml-[6%] t-border-solid t-border-2 t-border-slate-200 t-rounded-lg t-mt-4 t-px-2.5 t-pt-8 t-pb-6">
+    <div className="panel t-flex t-flex-col t-w-[600px] t-mr-6 t-border-solid t-border-2 t-border-slate-200 t-rounded-lg t-mt-4 t-px-2.5 t-pt-8 t-pb-6">
+      <PanelTopBar panelId={panelId} />
```

Comment:
After looking closely at this code, we can see that the panelId is being passed in every child component individually. I don't think that this is a good pattern. Maybe we can make use of the Context Provider by just keeping the panelId there and letting the components get it. This way we can get rid of all panelId props.